packages/dd-trace/src/appsec/rasp.js

```diff
@@ -3,18 +3,43 @@
 const { storage } = require('../../../datadog-core')
 const addresses = require('./addresses')
 const { httpClientRequestStart } = require('./channels')
+const web = require('../plugins/util/web')
 const waf = require('./waf')
+const { getBlockingAction, block } = require('./blocking')
+
+class AbortError extends Error {
+  constructor (req, res, blockingAction) {
+    super('AbortError')
+    this.name = 'AbortError'
+    this.req = req
+    this.res = res
+    this.blockingAction = blockingAction
+  }
+}
+
+function handleUncaughtException (err) {
+  if (err instanceof AbortError) {
+    const { req, res, blockingAction } = err
+    block(req, res, web.root(req), null, blockingAction)
+  } else {
+    throw err
+  }
+}
 
 const RULE_TYPES = {
   SSRF: 'ssrf'
 }
 
 function enable () {
   httpClientRequestStart.subscribe(analyzeSsrf)
+
+  process.on('uncaughtException', handleUncaughtException)
```

Comment:
As we are sending the AbortError asynchronously, we can't subscribe and unsubscribe to `uncaughtException` each time that we block. 
This is the reason why I am thinking to go back to the approach of throwing exception instead of using fake `ClientRequest`